app/src/main/java/com/frinder/frinder/activity/BaseActivity.java

```diff
@@ -0,0 +1,43 @@
+package com.frinder.frinder.activity;
+
+import android.content.DialogInterface;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import com.frinder.frinder.utils.ConnectivityChangeReceiver;
+
+public class BaseActivity extends AppCompatActivity implements ConnectivityChangeReceiver.OnConnectivityChangedListener{
+
+    private ConnectivityChangeReceiver connectivityChangeReceiver;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        connectivityChangeReceiver = new ConnectivityChangeReceiver(this);
+        IntentFilter filter = new IntentFilter();
+        filter.addAction("android.net.conn.CONNECTIVITY_CHANGE");
+        registerReceiver(connectivityChangeReceiver, filter);
+    }
+
+    @Override
+    public void onConnectivityChanged(boolean isConnected) {
+        if(!isConnected) {
+            final android.app.AlertDialog.Builder alertDialog = new android.app.AlertDialog.Builder(this);
+            alertDialog.setMessage("You don't have internet connection! Please enable and restart the app").setNegativeButton("Quit", new DialogInterface.OnClickListener() {
```

Comment:
Is there a way to ask to user to enable Internet and then we automatically refresh the page the user was trying to view?

Reply:
i tried looking for that the issue is we could have internet in 2 ways...either LTE/4G or WIFI...now which one would we prompt the user to enable? since we arent sure abt that?  I checked fb and yelp..they just say no internet connection ...what we could do is just show a snackbar instead and action as quit...do u want me to change that?

Reply:
changed it to Snackbar
![image](https://user-images.githubusercontent.com/4748819/31986350-048f52a0-b91d-11e7-9ed8-19af7de05e4d.png)